sdks/python/scripts/run_integration_test.sh

```diff
@@ -0,0 +1,180 @@
+#                      dirived from it including output, staging_location
+#                      and temp_location.
+#     sdk_location  -> Python tar ball location. Glob is accepted.
+#     num_workers   -> Number of workers.
+#     sleep_secs    -> Number of seconds to wait before verification.
+#     pipeline_opts -> List of space separateed pipeline options. If this
+#                      flag is specified, all above flag will be ignored.
+#                      Please include all required pipeline options when
+#                      using this flag.
+#
+# Test related flags:
+#     test_opts     -> List of space separated options to configure Nose test
+#                      during execution. Commonly used options like `--attr`,
+#                      `--tests`, `--nologcapture`. More can be found in
+#                      https://nose.readthedocs.io/en/latest/man.html#options
+#
+# Example usages:
+#     - Run full set of PostCommit tests with default pipeline options:
+#     `$ ./run_integration_test.sh`
+#
+#     - Run single integration test with default pipeline options:
+#     `$ ./run_integration_test.sh --test_opts --tests=apache_beam.examples.wordcount_it_test:WordCountIT.test_wordcount_it`
+#
+#     - Run full set of PostCommit tests with customized pipeline options:
+#     `$ ./run_integration_test.sh --project my-project --gcs_location gs://my-location`
+
+###########################################################################
+# Get pipeline options specified from commandline arguments.
+
+# Default pipeline options
+PROJECT=apache-beam-testing
+RUNNER=TestDataflowRunner
+GCS_LOCATION=gs://temp-storage-for-end-to-end-tests
+SDK_LOCATION=dist/apache-beam-*.tar.gz
+NUM_WORKERS=1
+SLEEP_SECS=20
+
+# Default test (nose) options.
+# Default test sets are full integration tests.
+TEST_OPTS="--attr=IT --nocapture"
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+case $key in
+    --runner)
+        RUNNER="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --project)
+        PROJECT="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --gcs_location)
+        GCS_LOCATION="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --sdk_location)
+        SDK_LOCATION="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --num_workers)
+        NUM_WORKERS="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --sleep_secs)
+        SLEEP_SECS="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --pipeline_opts)
+        PIPELINE_OPTS="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --test_opts)
+        TEST_OPTS="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    *)    # unknown option
+        echo "Unknown option: $1"
+        shift # past argument
```

Comment:
For my learning, do you not need to use another shift to pass value for this unknown argument?

Reply:
Depends on what format the unknown option is. It could be `--key` or `--key value`. what's the best way to handle them? Probably we can exit in this case to make a clear warning to user.

Reply:
I think an echo + exit in this case would be clearer than proceeding with a likely mistyped flag